docs/features/incremental-generators.md

```diff
@@ -42,154 +42,711 @@ public class MyGenerator : IIncrementalGenerator { ... }
 An assembly can contain a mix of diagnostic analyzers, source generators and
 incremental generators.
 
-### Initialization
-
-`IIncrementalGenerator` has an `Initialize` method that is called by the host
-(either the IDE or the command-line compiler) exactly once, regardless of the
-number of further compilations that may occur. `Initialize` passes an instance
-of `IncrementalGeneratorInitializationContext` which can be used by the
-generator to register a set of callbacks that affect how future generation
-passes will occur.
-
-Currently `IncrementalGeneratorInitializationContext` supports two callbacks:
+### Pipeline based execution
 
-- `RegisterForPostInitialization(GeneratorPostInitializationContext)`: the same
-  callback in shape and function as supported by Source Generators today
-- `RegisterExecutionPipeline(IncrementalGeneratorPipelineContext)`: replaces Execute, described below
+`IIncrementalGenerator` has an `Initialize` method that is called by the host[^1]
+exactly once, regardless of the number of further compilations that may occur.
 
-### Pipeline based execution
+[^1]: Such as the IDE or the command-line compiler
 
 Rather than a dedicated `Execute` method, an Incremental Generator instead
-creates an execution 'pipeline' as part of the initialization process via the
-`RegisterExecutionPipeline` method:
+defines an immutable execution pipeline as part of initialization. The
+`Initialize` method receives an instance of
+`IncrementalGeneratorInitializationContext`which is used by the generator to
+define a set of transformations:
 
 ```csharp
 public void Initialize(IncrementalGeneratorInitializationContext initContext)
 {
-    initContext.RegisterExecutionPipeline(context =>
-    {
-        // build the pipeline...
-    });
+    // define the execution pipeline here via transformations of initContext
```

Comment:
Consider a sample here where the `initContext` is used to access `SyntaxNode` or another item. Don't need a full generator but a quick and simple "here is how the data is accessed"

Reply:
Added an upfront example.